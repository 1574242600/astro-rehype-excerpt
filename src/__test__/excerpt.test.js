import { join, resolve } from 'path'
import { unified } from 'unified'
import remarkParse from 'remark-parse'
import remarkRehype from 'remark-rehype'
import rehypeRaw from 'rehype-raw'
import rehypeStringify from 'rehype-stringify'
import { read } from 'to-vfile'
import rehypeExcerpt from '../excerpt'

const processFixture = async (name, options) => {
    const path = join(resolve(), '/src/__test__/fixtures', name)
    const file = await read(path)
    const result = await unified()
        .use(remarkParse)
        .use(remarkRehype, { allowDangerousHtml: true })
        .use((opt) => (tree) => rehypeExcerpt(tree, opt), options)
        .use(rehypeRaw)
        .use(rehypeStringify, { allowDangerousHtml: true })
        .process(file)

    return result.toString()
}

test('should return unmodified document when no excerpt exists', async () => {
    const result = await processFixture('no-excerpt.md')

    expect(result).toMatchSnapshot()
})

test('should return excerpt using "exceprt" identifier', async () => {
    const result = await processFixture('excerpt.md')

    expect(result).toMatchSnapshot()
})

test('should return excerpt using "more" identifier', async () => {
    const result = await processFixture('more.md')

    expect(result).toMatchSnapshot()
})

test('should return excerpt using "preview" identifier', async () => {
    const result = await processFixture('preview.md')

    expect(result).toMatchSnapshot()
})

test('should return excerpt using "teaser" identifier', async () => {
    const result = await processFixture('teaser.md')

    expect(result).toMatchSnapshot()
})

test('should return excerpt using custom identifier', async () => {
    const options = {
        identifier: 'custom',
    }
    const result = await processFixture('custom.md', options)

    expect(result).toMatchSnapshot()
})

test('should return after excerpt when multiple exist', async () => {
    const result = await processFixture('multiple.md')

    expect(result).toMatchSnapshot()
})

test('should handle identifier with dashes', async () => {
    const options = {
        identifier: 'custom-with--dashes',
    }
    const result = await processFixture('custom-with-dashes.md', options)

    expect(result).toMatchSnapshot()
})

test('should handle identifier with spaces', async () => {
    const options = {
        identifier: 'custom with spaces',
    }
    const result = await processFixture('custom-with-spaces.md', options)

    expect(result).toMatchSnapshot()
})

test('should handle no spacing in exceprt comment', async () => {
    const result = await processFixture('no-space-in-comment.md')

    expect(result).toMatchSnapshot()
})

test('should handle mdx', async () => {
    const result = await processFixture('mdx.mdx')

    expect(result).toMatchSnapshot()
})