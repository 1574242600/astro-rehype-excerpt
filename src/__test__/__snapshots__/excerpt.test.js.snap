// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle identifier with dashes 1`] = `
"<h1>Title</h1>
<p>Paragraph 1.</p>
<p>Paragraph 2.</p>
"
`;

exports[`should handle identifier with spaces 1`] = `
"<h1>Title</h1>
<p>Paragraph 1.</p>
<p>Paragraph 2.</p>
"
`;

exports[`should handle mdx 1`] = `
"<p>import something from 'something';</p>
<p>export const meta = {
prop: 'value',
};</p>
<h1>Title</h1>
<p>Paragraph 1.</p>
<p>Paragraph 2.</p>
"
`;

exports[`should handle no spacing in exceprt comment 1`] = `
"<h1>Title</h1>
<p>Paragraph 1.</p>
<p>Paragraph 2.</p>
"
`;

exports[`should return after excerpt when multiple exist 1`] = `
"<h1>Title</h1>
<p>Paragraph 1.</p>
"
`;

exports[`should return excerpt using "exceprt" identifier 1`] = `
"<h1>Title</h1>
<p>Paragraph 1.</p>
<p>Paragraph 2.</p>
"
`;

exports[`should return excerpt using "more" identifier 1`] = `
"<h1>Title</h1>
<p>Paragraph 1.</p>
<p>Paragraph 2.</p>
"
`;

exports[`should return excerpt using "preview" identifier 1`] = `
"<h1>Title</h1>
<p>Paragraph 1.</p>
<p>Paragraph 2.</p>
"
`;

exports[`should return excerpt using "teaser" identifier 1`] = `
"<h1>Title</h1>
<p>Paragraph 1.</p>
<p>Paragraph 2.</p>
"
`;

exports[`should return excerpt using custom identifier 1`] = `
"<h1>Title</h1>
<p>Paragraph 1.</p>
<p>Paragraph 2.</p>
"
`;

exports[`should return unmodified document when no excerpt exists 1`] = `
"<h1>Title</h1>
<p>Paragraph 1.</p>
<p>Paragraph 2.</p>
<p>Paragraph 3.</p>
<p>Paragraph 4.</p>"
`;
